src/main/java/com/google/sps/data/NonPersistentMatchRepository.java

```diff
@@ -31,42 +31,61 @@ public User addTestData() {
     matchD.addSpecialty("DoS");
     matchD.addSpecialty("Security");
     User testUser = new User("Test User");
+    testUser.setId("000");
     List<User> allMatches = new ArrayList<User>(Arrays.asList(matchA, matchB, matchC, matchD));
-    userMatches.put(testUser, allMatches);
+    userMatches.put(testUser.getId(), allMatches);
     return testUser;
   }
 
-  public void addMatch(User user, User match) {
-    if (userMatches.containsKey(user)) {
-      List<User> currentMatches = userMatches.get(user);
+  //add a new match given User ID
+  public void addMatch(String userId, User match) {
+    if (userMatches.containsKey(userId)) {
+      List<User> currentMatches = userMatches.get(userId);
       currentMatches.add(match);
-      userMatches.replace(user, currentMatches);
+      userMatches.replace(userId, currentMatches);
     } else {
       List<User> newMatch = new ArrayList<User>(Arrays.asList(match));
-      userMatches.put(user, newMatch);
+      userMatches.put(userId, newMatch);
     }
   }
 
-  public void removeMatch(User user, User match) throws Exception {
-    if (userMatches.containsKey(user)) {
-      List<User> currentMatches = userMatches.get(user);
+  //add a new match given User
+  public void addMatch(User user, User match) {
```

Comment:
Do you still expect to need the addMatch(User user, User match) and removeMoatch(User user, User match) methods?

Reply:
I thought that it would be useful to have both in case one is easier for a certain implementation. I also use both in testing, but I could change it if you think it is unnecessary. Let me know!

Reply:
Okay - it seems like these might be removed eventually if you switch to storing user ids instead of Users or if you deprecate the NonPersistenetMatchRepository. Feel free to leave this as is.

Reply:
Sounds good.